Libraries/LibC/stdlib.cpp

```diff
@@ -104,9 +104,34 @@ static inline T strtol_impl(const char* nptr, char** endptr, int base)
     return num;
 }
 
+[[nodiscard]] bool __generate_unique_filename(char* pattern)
+{
+    int length = strlen(pattern);
+
+    if (length < 6 || !String(pattern).ends_with("XXXXXX")) {
+        errno = EINVAL;
```

Comment:
Hm, I wonder if we aren't clobbering `errno` in the `String` destructor here.
You're just moving the code around, so it's not your fault, but it does seem like a bad pattern.

Reply:
Huh didn't even consider that. Classic C++ trickery showing how little I know the language

I guess this is a good case to just use memcmp(). Would likely be faster and we wouldn't have to worry about those pesky C++ non-trivial types